rskj-core/src/main/java/co/rsk/pcc/blockheader/GetBitcoinHeader.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.pcc.blockheader;
+
+import co.rsk.pcc.ExecutionEnvironment;
+import co.rsk.pcc.NativeMethod;
+import org.ethereum.core.Block;
+import org.ethereum.core.CallTransaction;
+import org.ethereum.util.ByteUtil;
+
+import java.math.BigInteger;
+
+/**
+ * This implements the "getBitcoinHeader" method
+ * that belongs to the BlockHeaderContract native contract.
+ *
+ * @author Diego Masini
+ */
+public class GetBitcoinHeader extends NativeMethod {
+    private final BlockAccessor blockAccessor;
+
+    private final CallTransaction.Function function = CallTransaction.Function.fromSignature(
+            "getBitcoinHeader",
+            new String[]{"uint256"},
+            new String[]{"bytes"}
+    );
+
+    public GetBitcoinHeader(ExecutionEnvironment executionEnvironment, BlockAccessor blockAccessor) {
+        super(executionEnvironment);
+        this.blockAccessor = blockAccessor;
+    }
+
+    @Override
+    public CallTransaction.Function getFunction() {
+        return function;
+    }
+
+    @Override
+    public Object execute(Object[] arguments) {
+        short blockDepth = ((BigInteger) arguments[0]).shortValueExact();
```

Comment:
Using `shortValueExact` means that you only allow numbers up to `32767`, or else you throw an exception. This doesn't match with:

* the expected behavior of the contract (returning an empty byte array).
* the expected behavior of the `uint256` solidity type

Reply:
This is tricky because we should allow values up to `4294967295 (0xffffffff)`. I wrote a `Uint24` class as part of the `unitrie` work, maybe we can have a `Uint256` class too.

Reply:
We want to limit the depth to 5000, which is the number of blocks that fits in the IndexedBlockStore cache. The problem you describe will occur with any uint value we select, being uint16 the closest solidity type to represent the maximum depth. In Java the closest numeric type to represent the maximum depth is short.

Reply:
Let's say the user passes `MAX_SHORT + 1` to this method. It will fail with an exception instead of returning an empty byte array.

Is this how the method is supposed to work?